README.md
# Custom Maps
> Control options and customization of the Google Maps colors, just 1,7 KB.

<center>
  <img src="preview.png" alt="Preview">
</center>

### Install

```bash
yarn add cmaps
```

After installing, include the scripts in html.

```html
<script src="node_modules/cmaps/dist/cMaps.min.js"></script>
```

### Wrapper map need be something like:

```html
<div id="map"></div>
```

### Starting map
```js
var map = new cMaps(options);
```

### Options

Option | Type | Description | Example    | Required
------ | ---- | ----------- | -----------|------------
wrapperId | string | Container id map | 'map' | true
Key | string | Google API Key | ['api_key'](https://developers.google.com/maps/documentation/javascript/get-api-key) | true
pinLat | number | pin latitude | -23.562353 | true
pinLng | number | pin latitude | -46.503113 | true
mapLat | number | map latitude | -23.562353 | true
mapLng | number | map latitude | -46.503113 | true
styles | array | google styles | [{},{}] | true
name | string | Map name | 'cMaps' | false
icon | string | url to pin the way | 'imgs/pin.png' | false
zoom | number | Initial zoom map | 16 | false
zoomControl | boolean | Manual zoom control | true | false
scrollwheel | boolean | Mouse scroll for zoom | false | false
mapTypeControl | boolean | Choose the map type in the upper left corner | true | false
draggable | boolean | If this option is false the map can not be dragged | true | false
streetView | boolean | Control street view | false | false


### Usage

```js
// *Example* Code that was exported from Styled Maps Wizard or Snazzy Maps
var styles = [{"featureType":"all","elementType":"labels","stylers":[{"visibility":"on"}]} ...];

var map = new cMaps({
  'name': 'cMap',
  'wrapperId':'map',
  'pinLat': -23.562353,
  'pinLng': -46.503113,
  'mapLat': -23.562353,
  'mapLng': -46.503113
  'styles': styles,
  'key': 'YOUR_API_KEY'
});
```

#### Style

Use the [Styled Maps Wizard](http://googlemaps.github.io/js-samples/styledmaps/wizard/index.html) or [Snazzy Maps](https://snazzymaps.com) to export a JSON with all possible options and include as a parameter to the cMap.

## Example

You can see this example in `localhost:8080`, enter in example directory:

```bash
yarn start
```

#### Disclaimer
> In the example folder has an demonstration of everything working if necessary :)

CHANGELOG.md
# Change Log
All notable changes to this project will be documented in this file.
This project adheres to [Semantic Versioning](http://semver.org/).

## [3.0.0] - 2017-12-28

### Changed
- Name of package
- register package in npmjs.org

## [2.0.4] - 2017-11-15

### Changed
- duplicate api key in multiples map instances

## [2.0.3] - 2017-11-02

### Changed
- Update to ES6

### Added
- Babeljs

### removed
- Gulpjs

## [2.0.2] - 2017-11-01

### Changed
- Fix README.md

## [2.0.1] - 2017-11-01

### Changed
- change icon example

## [2.0.0] - 2017-11-01

### Added
- Support modules

### removed
- Color options

### Changed
- properties lat and lang that were in markup are now passed as parameter when cMap is started
- style is now a property of options
- dynamically creating google API script

## [1.0.1] - 2016-10-01
### Added
- Style option to [Snazzy Maps](https://snazzymaps.com).
- Download option for bower.

### Changed
- Name of the main file map.js to cMap.js

## [1.0.0] - 2016-07-30
### Added
- Google maps and standard control over the map features such as zoom, google street view, etc.
- Basic Options Map stylization.
- Integration with json files [Styled Maps Wizard](http://googlemaps.github.io/js-samples/styledmaps/wizard/index.html)


app/scripts/main.js
(function(){

  var styles = [{"featureType":"all","elementType":"labels","stylers":[{"visibility":"on"}]},{"featureType":"all","elementType":"labels.text.fill","stylers":[{"saturation":36},{"color":"#000000"},{"lightness":40}]},{"featureType":"all","elementType":"labels.text.stroke","stylers":[{"visibility":"on"},{"color":"#000000"},{"lightness":16}]},{"featureType":"all","elementType":"labels.icon","stylers":[{"visibility":"off"}]},{"featureType":"administrative","elementType":"geometry.fill","stylers":[{"color":"#000000"},{"lightness":20}]},{"featureType":"administrative","elementType":"geometry.stroke","stylers":[{"color":"#000000"},{"lightness":17},{"weight":1.2}]},{"featureType":"administrative.country","elementType":"labels.text.fill","stylers":[{"color":"#e5c163"}]},{"featureType":"administrative.locality","elementType":"labels.text.fill","stylers":[{"color":"#c4c4c4"}]},{"featureType":"administrative.neighborhood","elementType":"labels.text.fill","stylers":[{"color":"#e5c163"}]},{"featureType":"landscape","elementType":"geometry","stylers":[{"color":"#000000"},{"lightness":20}]},{"featureType":"poi","elementType":"geometry","stylers":[{"color":"#000000"},{"lightness":21},{"visibility":"on"}]},{"featureType":"poi.business","elementType":"geometry","stylers":[{"visibility":"on"}]},{"featureType":"road.highway","elementType":"geometry.fill","stylers":[{"color":"#e5c163"},{"lightness":"0"}]},{"featureType":"road.highway","elementType":"geometry.stroke","stylers":[{"visibility":"off"}]},{"featureType":"road.highway","elementType":"labels.text.fill","stylers":[{"color":"#ffffff"}]},{"featureType":"road.highway","elementType":"labels.text.stroke","stylers":[{"color":"#e5c163"}]},{"featureType":"road.arterial","elementType":"geometry","stylers":[{"color":"#000000"},{"lightness":18}]},{"featureType":"road.arterial","elementType":"geometry.fill","stylers":[{"color":"#575757"}]},{"featureType":"road.arterial","elementType":"labels.text.fill","stylers":[{"color":"#ffffff"}]},{"featureType":"road.arterial","elementType":"labels.text.stroke","stylers":[{"color":"#2c2c2c"}]},{"featureType":"road.local","elementType":"geometry","stylers":[{"color":"#000000"},{"lightness":16}]},{"featureType":"road.local","elementType":"labels.text.fill","stylers":[{"color":"#999999"}]},{"featureType":"transit","elementType":"geometry","stylers":[{"color":"#000000"},{"lightness":19}]},{"featureType":"water","elementType":"geometry","stylers":[{"color":"#000000"},{"lightness":17}]}];

  var map = new cMaps({
    'name': 'cMaps',
    'wrapperId':'example',
    'pinLat': -23.562353,
    'pinLng': -46.503113,
    'mapLat': -23.562353,
    'mapLng': -46.503113,
    'zoom': 10,
    'icon': 'app/images/university.svg',
    'mobile': false,
    'zoomControl': true,
    'mapTypeControl': true,
    'scrollwheel': false,
    'streetView': true,
    'styles': styles,
    'key': 'AIzaSyCP3iIRdwkWKwegt2dU4nA4KdpiInTmSlE'
  });

})();
